src/util/Options.js

```diff
@@ -132,20 +140,37 @@ class Options extends null {
   }
 
   /**
-   * Create a cache factory using predefined limits.
-   * @param {Record<string, number>} [limits={}] Limits for structures.
+   * Create a cache factory using predefined settings to sweep or limit.
+   * @param {Record<string, SweptCollectionOptions|number>} [settings={}] Settings passed to the relevant constructor.
+   * If no setting is provided for a manager, it uses Collection.
    * @returns {CacheFactory}
+   * @example
+   * Options.cacheSome({
+   *    MessageManager: 200,
+   *    ThreadManager: {
+   *      sweepInterval: SweptCollection.filterByLifetime({
+   *        getComparisonTimestamp: e => e.archiveTimestamp,
+   *        excludeFromSweep: e => !e.archived,
+   *      }),
+   *    },
+   *  });
    */
-  static cacheWithLimits(limits = {}) {
+  static cacheSome(settings = {}) {
     const Collection = require('./Collection');
     const LimitedCollection = require('./LimitedCollection');
+    const SweptCollection = require('./SweptCollection');
 
     return manager => {
-      const limit = limits[manager.name];
-      if (limit === null || limit === undefined || limit === Infinity) {
+      const setting = settings[manager.name];
+      if (typeof setting === 'number' && setting !== Infinity) return LimitedCollection(setting);
+      if (
+        setting === null ||
+        typeof setting !== 'object' ||
+        (setting.sweepInterval ? setting.sweepInterval <= 0 || setting.sweepInterval === Infinity : true)
```

Comment:
This can just be `setting?.sweepInterval == null || setting.sweepInterval <= 0 || setting.sweepInterval === Infinity`.